src/routes/api-routes.js

```diff
@@ -0,0 +1,219 @@
+    var len = dummy[stream_id].length
+    dummy[stream_id].events[len] = { index: len, data: data }
+    dummy[stream_id].length += 1
+    success = true
+  }
+  return success
+}
+
+// =============== Route URLs =============== //
+const length_url = api_base_url + '/streams/:stream_id'
+const load_url = api_base_url + '/streams/:stream_id/events/:from/:to'
+const push_url = api_base_url + '/streams/:stream_id/events'
+
+// =============== Route Schemas =============== //
+const length_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        stream: { type: 'string' },
+        length: { type: 'integer' },
+        msg:  { type: 'string' },
+      },
+    }
+  }
+}
+
+const load_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        stream: { type: 'string' },
+        events: {
+          type: 'array',
+          items: {
+            type: 'object',
+            properties: {
+              index: {type: 'integer'},
+              data: {type: 'string'},
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+const push_schema =
+{
+  response: {
+    200: {
+      type: 'object',
+      properties: {
+        success: {type: 'boolean'},
+        msg: {type: 'string'},
+      },
+    },
+    201: {
+      type: 'object',
+      properties: {
+        success: {type: 'boolean'},
+        msg: {type: 'string'},
+      },
+    },
+  },
+}
+
+// =============== Route Handlers =============== //
+async function length_handler (req, res) {
+  const stream_id = req.params.stream_id
+  const len = await length(stream_id)
+  const error_msg = (len === null) ? 'stream_id not found' : '';
+
+  const resp =
+  {
+    stream: stream_id,
+    length: len,
+    msg : error_msg,
+  }
+
+  return JSON.stringify(resp)
+}
+
+async function load_handler (req, res) {
+  const stream_id = req.params.stream_id
+  const from = req.params.from
+  const to = req.params.to
+
+  const events_array = await load(stream_id, from, to)
+
+  const resp = {
+    stream: stream_id,
+    events: events_array,
+  }
+
+  return JSON.stringify(resp)
+}
+
+async function push_handler (req, res) {
+  const stream_id = req.params.stream_id
+  const data = req.body.data
+  const is_successful = await push(stream_id, data)
+  var msg = "OK"
+  if (!is_successful) {
+    res.code(201)
+    msg = `An error occured when trying to push data to stream ${stream_id}`
+  }
+
+  const resp = {
+    success: is_successful,
+    msg: msg
+  }
+
+  return resp
+}
+
+// =============== Routes =============== //
+const routes = [
+  // length
+  {
+    method: 'GET',
+    url: length_url,
+    schema: length_schema,
+    handler: length_handler
+  },
+
+  // load
+  {
+    method: 'GET',
+    url: load_url,
+    schema: load_schema,
+    handler: load_handler
+  },
+
+  // push
+  {
+    method: 'POST',
+    url: push_url,
+    schema: push_schema,
+    handler: push_handler
+  }
+]
+
+module.exports = routes
```

Comment:
In general, I think we can rename things to more "REST" names: `get_stream_events`, `get_stream` and `push_event`.